src/script/hooks/useDebounce.test.tsx

```diff
@@ -0,0 +1,97 @@
+/*
+ * Wire
+ * Copyright (C) 2023 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import {renderHook, act} from '@testing-library/react';
+
+import {useDebounce} from './useDebounce';
+
+describe('useDebounce', () => {
+  beforeAll(() => {
+    jest.useFakeTimers();
+  });
+
+  afterAll(() => {
+    jest.useRealTimers();
+  });
+
+  it('should debounce the callback', () => {
+    const callback = jest.fn();
+    const time = 1000;
+
+    renderHook(() => useDebounce(callback, time));
+
+    act(() => {
+      jest.advanceTimersByTime(time - 1);
+    });
+    expect(callback).not.toHaveBeenCalled();
+
+    act(() => {
+      jest.advanceTimersByTime(1);
+    });
+    expect(callback).toHaveBeenCalled();
+  });
+
+  it('should only call the callback once when multiple calls are made within the debounce time', () => {
+    const callback = jest.fn();
+    const time = 1000;
+
+    renderHook(() => useDebounce(callback, time));
+
+    act(() => {
+      jest.advanceTimersByTime(200);
+      jest.advanceTimersByTime(800);
+      jest.advanceTimersByTime(200);
+    });
+    expect(callback).toHaveBeenCalledTimes(1);
+  });
+
+  it('should not call the callback if it is unmounted before the debounce time', () => {
+    const callback = jest.fn();
+    const time = 1000;
+
+    const {unmount} = renderHook(() => useDebounce(callback, time));
+
+    act(() => {
+      unmount();
+      jest.advanceTimersByTime(500);
```

Comment:
Shouldn't this time be more than 1000 here? To make sure callback would run if component wasn't unmounted? This tests will pass even when your remove `unmount()` from here.

Reply:
I got your point but this test is - should not call the callback if it is unmounted before the debounce time

we need to make it - should not call the callback if it is unmounted after the debounce time

Reply:
actually before/after time doesn't matter. if the component is unmounted before you advanced the timer to something > 1000 / <1000 the callback won't be called. so the important thing here is the component is unmounted before the timer could finish

Reply:
`should not call the callback if it is unmounted before the debounce time` callback will never be called in this case, doesn't matter if we unmount or not. So this is exactly the same thing you're testing in the first case - that callback should not run before debounce time passes

Reply:
I think it makes more sense to advance the timer above 1000 to make the purpose of the test more clear